src/labs.txt

```diff
@@ -2745,6 +2745,200 @@ Output:
 =hist
 EOF
 
+----------------------------------------------------------------------
+h1. Finding Commits That Introduced a Change
+
+h2. Goals
+
+* Learn how to use @git bisect@ to find a commit that introduced a change.
+
+p. It is sometimes necessary to find a commit that introduced a specific
+change.  Commonly you will be searching for a bug, but sometimes the change
+you're looking for is a feature. Git allows you to efficiently search for a
+commit as long as you can describe your search criteria in terms of a command
+to run.
+
+p. In this lab we'll be searching for the commit that allowed users to specify
+their name. While you could find it manually in this project, when there are
+hundreds or thousands of commits this becomes impractical. @git bisect@ will
+make this task trivial.
+
+h2. The Test Command
+
+p. The first thing we need is a test command. @git bisect@ works by checking
```

Comment:
You can give bisect a script to automate it and it just uses exit 0 or exit non-0 for bad/good or old/new respectively.

Reply:
I follow that, my understanding/assumption is that the more common usage pattern is for people to manually declare good/bad and not write a script.  I know I personally have only ever written a script runner probably once or twice ever and I've never taught it that way as the "default" to people

Reply:
I agree with Craig here.  I would still like to keep the script content, as it is awesome...  it's just better as a "you can also do this kickass thing to automate the manual thing we just showed you" add-on topic.

Reply:
Interesting. I write a script for almost all of my bisects. I will point out (and I understand if it's hard to see because there's no preview) that the lab does start by running the script manually and doing `git bisect old/new`. If you think it's better to leave the script off entirely until the automation step I can do that too, though I do like the idea that it encourages getting the computer to do the work for you.

Reply:
I'd mention the script and `bisect run` option as a detail, but the more fundamental technique is just without a script. It's often difficult to write a script for what you're looking for.  "Good" vs. "Bad" might be a visual regression, for example.